### PR TITLE
fixed regression: reimplemented warhammer 2 wsmodel generator

### PR DESCRIPTION
**Pull request by admiralnelson**
Hi, ole. 
I have implemented wsmodel generator for warhammer 2.

### CODE REVIEW

File: View3D/Services/WsModelGeneratorService.cs [hunk 1]
```diff
@@ -78,6 +81,69 @@ public void GenerateWsModel()
             }
         }
 
+        public void GenerateWsModelForWh2()
```
Comment (olekristianhomelien) [inline View3D/Services/WsModelGeneratorService.cs:84]: This function should be pretty much the same for wh2 and wh3. Need to keep code duplication low

File: View3D/Services/WsModelGeneratorService.cs [hunk 2]
```diff
@@ -78,6 +81,69 @@ public void GenerateWsModel()
...
+                SaveHelper.Save(_packFileService, wsModelPath, existingWsModelFile, Encoding.UTF8.GetBytes(wsModelData));
+            }
+            catch (Exception e)
+            {
+                _logger.Here().Error("Error generating ws model - " + e.Message);
+                MessageBox.Show("Generation failed!");
+            }
+        }
+
+        string GenerateWsModelForWh2(string modelFilePath, bool onlyVisible)
```
Comment (olekristianhomelien) [inline View3D/Services/WsModelGeneratorService.cs:120]: This function should be pretty much the same for wh2 and wh3. Need to keep code duplication low

File: View3D/Services/WsModelGeneratorService.cs [hunk 3]
```diff
@@ -144,6 +217,13 @@ string LoadMaterialTemplate()
             var result = reader.ReadToEnd();
             return result;
         }
+        string LoadMaterialTemplateWh2()
```
Comment (olekristianhomelien) [inline View3D/Services/WsModelGeneratorService.cs:220]: I think the only difference in code should be the materials. Maybe have two MaterialGenerator classes, and instanciate the correct one based on an input enum (GameType, cant remember what is actually called) Throw exception if game not supported

**Review by olekristianhomelien — CHANGES_REQUESTED** [review]

File: View3D/Services/WsModelGeneratorService.cs (hunk 3 above)
Comment (admiralnelson) [reply View3D/Services/WsModelGeneratorService.cs:220]: OK, noted. I will push another changes.

**Pull request merged**